querybook/server/lib/event_logger/base_event_logger.py

```diff
@@ -22,3 +79,41 @@ def log(self, uid: int, event_type: EventType, event_data: dict) -> None:
             event_data (dict): addtional info of the event in JSON format.
         """
         raise NotImplementedError()
+
+    def log_api_request(self, uid: int, route: str, method: str, params: dict) -> None:
+        """Log an API request.
+
+        Args:
+            method (str): request method, e.g. GET, POST
+            route (str): route of the api endpoint which serves the request
+            params (dict): params of the request, includes path params,
+                query strings and post body
+        """
+        if not self.should_log_api_request(route, method):
+            return
+
+        params = self.__prune_api_request_params(params)
+        event_data = {"method": method, "route": route, "params": params}
+        self.log(uid=uid, event_type=EventType.API, event_data=event_data)
+
+    def __prune_api_request_params(self, params: dict):
```

Comment:
by convention, i think __ means its private right? so should_log_api_request and api_block_list should be protected in this case with a `_` at front?